### ISSUE TITLE
[1.10-rc1] `docker pull nonexistent-or-private-image` should not say "unauthorized"

### ISSUE THREAD

**Issue by tiborvass** [docker/docker#19419]
With 1.9:
```
$ docker-1.9 pull foobar
Using default tag: latest
Pulling repository docker.io/library/foobar
Error: image library/foobar:latest not found
```

With 1.10.0-rc1:
```
$ docker-1.10.0-rc1 pull foobar
Using default tag: latest
Pulling repository docker.io/library/foobar
unauthorized: access to the requested resource is not authorized
Error: image library/foobar not found
```

I'd expect the output to not say "unauthorized" since that's an implementation detail from docker UX's standpoint: https://github.com/docker/distribution/issues/1291

**Comment by aaronlehmann** [comment]
The daemon's behavior here isn't really any different than 1.9, it's just that 1.9 was hiding the first error. https://github.com/docker/docker/pull/17617 caused both errors to be visible. We can consider reverting #17617 if this is a problem.

**Comment by aaronlehmann** [comment]
But keep in mind that any cosmetic fix will have to be revisited for 1.11, which won't have v1 fallbacks masking the 401. It probably makes sense to pursue a registry-side fix in parallel.

**Comment by tiborvass** [comment]
@aaronlehmann thanks for the root cause. I'd suggest we revert the PR, and fix this registry-side before 1.11. Once that's done we can revert the revert?

**Comment by dmp42** [comment]
@tiborvass after a quick chat with @aaronlehmann : reverting #17617 as a way to fix cosmetics for 1.10 sounds ok.

Fixing the underlying issue should really be done for 1.11 though (cc @docker/distribution-maintainers ) - and this is probably tricky to get right.

**Comment by tiborvass** [comment]
@dmp42 ok thanks for confirming!
